### PR TITLE
parallelize and promise.all

### PR DESCRIPTION
**Pull request by kkoskelin**
When possible, replacing `for(...){ await X}` with parallel interactor calls.

### CODE REVIEW

**Review by rachaelparris — APPROVED** [review]

File: web-client/src/presenter/actions/DocketEntry/submitDocketEntryAction.js [hunk 1]
```diff
@@ -81,15 +80,17 @@ export const submitDocketEntryAction = async ({
...
-        applicationContext,
-        caseId: caseDetail.caseId,
-        documentId: document.documentId,
-      });
-    }
-  }
+  const pendingDocuments = caseDetail.documents.filter(
+    document => document.processingStatus === 'pending',
+  );
+  const createCoverSheetInteractor = async document => {
```
Comment (codyseibert) [inline web-client/src/presenter/actions/DocketEntry/submitDocketEntryAction.js]: I'm not sure if we need async / await for this function

File: web-client/src/presenter/actions/FileDocument/fileExternalDocumentAction.js [hunk 2]
```diff
@@ -66,15 +66,17 @@ export const fileExternalDocumentAction = async ({
...
-        caseId: caseDetail.caseId,
-        documentId: document.documentId,
-      });
-    }
-  }
+  const pendingDocuments = caseDetail.documents.filter(
+    document => document.processingStatus === 'pending',
+  );
+  const createCoverSheetInteractor = async document => {
+    return await applicationContext.getUseCases().createCoverSheetInteractor({
```
Comment (codyseibert) [inline web-client/src/presenter/actions/FileDocument/fileExternalDocumentAction.js]: same here

File: web-client/src/presenter/actions/FileDocument/submitCaseAssociationRequestAction.js [hunk 3]
```diff
@@ -95,15 +96,17 @@ export const submitCaseAssociationRequestAction = async ({
...
-        caseId: caseDetail.caseId,
-        documentId: document.documentId,
-      });
-    }
-  }
+  const pendingDocuments = caseDetail.documents.filter(
+    document => document.processingStatus === 'pending',
+  );
+  const createCoverSheetInteractor = async document => {
+    return await applicationContext.getUseCases().createCoverSheetInteractor({
```
Comment (codyseibert) [inline web-client/src/presenter/actions/FileDocument/submitCaseAssociationRequestAction.js]: and here

**Pull request merged**